### ISSUE TITLE
[v2.987Extra] Chromecast starts video but app gets stuck 

### ISSUE THREAD

**Issue by dagofr01** [SkyTubeTeam/SkyTube#1209]
start of video on Chromecast starts video successfully but the app gets stuck on the videolist with infinite spinner . No way to control video. 
Fairphone4 with current A12. 
Using Chromecast with other apps is no issue.

**Comment by sekian** [comment]
I have the same issue. I am using a Redmi Note 12 Pro with Android 13.

From my side, I have tried to cast using:
- version **2.987**: An **endless** spinning wheel appears (trying to load the controls?) while the cast video starts in the TV screen. Since the controls do not load, you have no control. Moreover, minimizing the app makes it fail and if you reopen the app it crashes instantly. The app will keep crashing on reopen until you stop the casting on the TV screen.
-  version **2.986**: The controls appear in the phone screen, but the video does not load, and the TV screen is black.

I was able to get a stack trace of the error after it crashes on reopen:
```java
java.lang.IllegalStateException: TypeToken must be created with a type argument: new TypeToken<...>() {}; When using code shrinkers (ProGuard, R8, ...) make sure that generic signatures are preserved.
 at com.google.gson.reflect.TypeToken.getTypeTokenTypeArgument(Unknown Source:40)
 at com.google.gson.reflect.TypeToken.<init>(SourceFile:0)
 at free.rm.skytube.gui.fragments.ChromecastBaseControllerFragment$1.<init>(Unknown Source:2)
 at free.rm.skytube.gui.fragments.ChromecastBaseControllerFragment.init(SourceFile:0)
 at free.rm.skytube.gui.fragments.ChromecastMiniControllerFragment.init(Unknown Source:0)
 at free.rm.skytube.gui.fragments.ChromecastBaseControllerFragment.init(SourceFile:0)
 at free.rm.skytube.gui.activities.BaseActivity$SessionManagerListenerImpl.lambda$onSessionResumed$0(Unknown Source:33)
 at free.rm.skytube.gui.activities.BaseActivity$SessionManagerListenerImpl.$r8$lambda$GfyoBHyqmZ6PfUeKOE-xixFt2mw(SourceFile:0)
 at free.rm.skytube.gui.activities.BaseActivity$SessionManagerListenerImpl$$ExternalSyntheticLambda0.run(SourceFile:0)
 at android.os.Handler.handleCallback(Handler.java:942)
 at android.os.Handler.dispatchMessage(Handler.java:99)
 at android.os.Looper.loopOnce(Looper.java:211)
 at android.os.Looper.loop(Looper.java:300)
 at android.app.ActivityThread.main(ActivityThread.java:8321)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:581)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1028)
```

**Comment by edmundlaugasson** [comment]
Having same issue with 2.987 Extra. App Will start video in Chromecast but SkyTube itself will crash. Still SkyTube keeps streaming to Chromecast while crashing itself.

**Comment by dagofr01** [comment]
If the video is finished one can go back to skytube and select again and it still is connected to the chromecast device.

**Comment by drogga** [comment]
Merge your last 2 comments if you want and remove `v2.987` from the title as it's no specific only to it and maybe mention/tag gzsombor if you want in the merged comment, not that I think will change anything...

Please don't mention me if you reply (no need to).

**Comment by drogga** [comment]
The version will eventually get bumped again (2.994 was pointless, given that F-Droid had no issues building it) and the issue will still be unresolved, that's why I suggested leaving only the mention of Extra there, now it's even worse and you didn't followed my suggestion regarding your last 2 comments...

**Comment by dagofr01** [comment]
@drogga I try to think first about what I am doing. I am not in a hurry.

**Comment by gzsombor** [comment]
I don't have any hardware that supports Chromecast, and I never had, so I've never tested it.
The exception looks strange, as it seems to be coming from Gson when it tries to convert a string into a video object - but the same code is there since 2018, when that code was introduced first, and it looks correct, so it must be some problem with the r8/proguard thingy - that also rarely touched.

@drogga : F-Droid could build 2.993 only after I've opened a separate PR to their repo and please stop mentioning me in every second of your comments and manipulating others to do the same, I've asked you to stop doing that long time ago. Your harassment was big part of cause that I've lost my enthusiasm for this project.

**Comment by drogga** [comment]
OK and I'm sorry you feel that way, but a 'harassment' & 'manipulation' - lol, my intentions are all good, I don't remember you ever telling me anything like this and I don't mention you every time, only when I have to an it seems that it makes sense, plus I suggested the OP here to do so seeing that this is almost 2 years old and I and the others reporting here would probably like it resolved, but how can we know that you've seen it or care, when there's no indication of you doing so (as comments, reactions, labels), mentioning someone, especially the maintainer(s) on GitHub and other alternative platforms is something common and normal and I now start to realize that probably it's something personal - you **usually** not responding to / ignoring my comments, you are actually hard to reach to outside from here, if not impossible at least for me, what I mean is that we can have hundreds of comments here and you still might not respond or do anything if there's a problem and there are many, it's not like there's another maintainer, I mean I suggested doing something to change this multiple times, but - nothing, I even planned to do some more PRs (mostly in the manifests), but seeing how you feel and that it might take months to get merged if at all and that I don't get advices from you on questions of how to handle some things and there are many that needs to be handled (history shows), means that whatever I do doesn't matter actually and that it's pointless for me to waste more of my time anymore. If I had the permissions you do here, I would handle myself some things I can't currently do as a contributor and won't bother you for some of the things I already did, if I don't ask you, then who ?, who can do them ?, there's no one else, do you see any PRs from someone else here in this repo ? - no, no other volunteers, nobody cares. Sorry amd apologies that I do actually care and don't want this app to be discontinued, but get better.

Now, as you can see I didn't mention you properly this time **ZSombor** on purpose and as you asked, but I will still appreciate if you react on that comment with the GH's dedicated feature if you read it, so I can delete it and not be such a burden and annoyance to you anymore. Thank You.
I will ask you to try to not make questionable changes anymore, like the new obtrusive overflow, pop-up or whatever is called subscribe button in channels, that needs fixing or reverting to how it was before.

**Comment by gzsombor** [comment]
Sorry, maybe I wasn't clear enough, when I said years ago, when you asked for having direct, private channel to communicate, I said that I don't want that, having this issue tracker is more than enough  and if you find a bug or an issue to discuss, open a ticket and if I have enough free time, maybe I could try to fix it. 
 For me, it's a feature, not a bug, that I don't read all the comment notifications from this project, it is intentional. Whenever Youtube changes, I tend to check the tickets and answer those for that I know the answer. If I don't know, or don't have the time to solve the ticket, then I don't do anything.
Unfortunately I don't have too much free time and I have other things to do than fixing SkyTube - if SkyTube actually works, I have even less time 😄 
 I totally understand your - and everyone else's frustrations - about the situation. But I never promised more, never promised to fix every bugs, never promised to not introduce more bugs, never promised to implement every idea, etc.

**Comment by dagofr01** [comment]
@gzsombor or may some one who was historically involved: 
if you have no chromecast for testing then who actually came up with the EXTRA versions which do seem to be primarily for this purpose and who was/is testing those?

**Comment by gzsombor** [comment]
It's implemented in this PR: https://github.com/SkyTubeTeam/SkyTube/pull/365

**Comment by dagofr01** [comment]
@gzsombor  You know what ? With v2.995Extra  the crash has vanished. I guess you made some major changes, may be that removed it coincidentally. Remote controlling chromecast is working again.

Anybody else ?

**Comment by gzsombor** [comment]
Thanks for reporting back, yes, I've refactored the JSON parsing in https://github.com/SkyTubeTeam/SkyTube/pull/1335 to fix this issue - but I couldn't test it.

**Comment by sekian** [comment]
I updated to v2.995 Extra and tested casting to an Android TV and to a Chromecast 3. I can confirm the issue is now fixed. Thank you.
